packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -705,6 +708,16 @@ class ScrollableState<T extends Scrollable> extends State<T> with SingleTickerPr
     });
   }
 
+  // Used for state changes that sometimes occur during a build phase. If so,
+  // we skip calling setState, as the changes will apply to the next build.
+  void _setStateMaybeDuringBuild(VoidCallback fn) {
+    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
```

Comment:
I'm not sure this is really valid. What if your ancestor is setting it on you during their build, for instance?

But I'm fine with checking this in. It will work around a bug that people are complaining about, and this code is all getting rewritten in my refactor anyway.

cc @abarth who may have opinions about this in particular.